lib/wallaroo/core/sink/connector_sink/connector_sink.pony

```diff
@@ -504,35 +556,57 @@ actor ConnectorSink is Sink
       // commit/abort decisions only on the range of output
       // governed by a single round of 2PC.
 
-      _phase = QueuingSinkPhase(_sink_id,
-        this)
+      try
```

Comment:
This is inside out for how we normally do the state transitions.  Can we add a method on `SinkPhase` that triggers `Fail()` on anything but `BarrierSinkPhase`, but on `BarrierSinkPhase` causes it to call a different method on `ConnectorSink` for the queuing transition, passing it's queue along?

Reply:
Fixed by 19276f379d5cbd2ae536a14e6aed37e72452f365